docs/_extension/component_directive.py

```diff
@@ -0,0 +1,92 @@
+"""A directive to generate the list of all the built-in components.
+
+Read the content of the component folder and generate a list of all the components.
+This list will display some informations about the component and a link to the
+GitHub file.
+"""
+import re
+from pathlib import Path
+from typing import Any, Dict, List
+
+from docutils import nodes
+from sphinx.application import Sphinx
+from sphinx.util import logging
+from sphinx.util.docutils import SphinxDirective
+
+logger = logging.getLogger(__name__)
+
+TEMPLATE_LIST = '<ul class="simple">{content}</ul>'
+TEMPLATE_LINE = '<li><a href="{url}">{name}</a>:  {description}</li>'
```

Comment:
It'd be fairly easy to express this in terms of nodes rather than pre-written HTML, if you'd like. Doing that would also have the benefit of eg internal/external link tracking, translations, etc.

Reply:
I would love to do that but failed to find examples. If you have any suggestion I'd be happy to follow them.

Reply:
Sonething like the below (untested) ought to work?
```python
item = nodes.list_item('', nodes.reference('', name, refuri=url), nodes.Text(f': {description}'))
nodes.bullet_list('', item, item2, item3, ...)
```